segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/OperationProcessor.java

```diff
@@ -437,7 +484,9 @@ int getPendingCount() {
          */
         void frameSealed(DataFrameBuilder.CommitArgs commitArgs) {
             synchronized (stateLock) {
-                commitArgs.setIndexKey(OperationProcessor.this.metadataUpdater.sealTransaction());
+                commitArgs.setMetadataTransactionId(OperationProcessor.this.metadataUpdater.sealTransaction());
+                commitArgs.setOperations(Collections.unmodifiableList(this.nextFrameOperations));
+                this.nextFrameOperations = new ArrayList<>(MAX_READ_AT_ONCE);
```

Comment:
This is allocating an array of 1000 every time. This seems excessive. If we can either use a single array without re-allocating or here and in the constructor not pass a size recommendation, it will lower the memory allocated here.

Reply:
If we use a single array then we need to make a copy somewhere. I prefer not to.

I removed the default size.